source/common/common/assert.cc

```diff
@@ -28,15 +28,63 @@ class ActionRegistrationImpl : public ActionRegistration {
   static std::function<void()> debug_assertion_failure_record_action_;
 };
 
+class EnvoyBugRegistrationImpl : public ActionRegistration {
+public:
+  EnvoyBugRegistrationImpl(std::function<void()> action) {
+    ASSERT(envoy_bug_failure_record_action_ == nullptr);
+    envoy_bug_failure_record_action_ = action;
+  }
+
+  ~EnvoyBugRegistrationImpl() override {
+    ASSERT(envoy_bug_failure_record_action_ != nullptr);
+    envoy_bug_failure_record_action_ = nullptr;
+  }
+
+  static bool shouldLogAndInvoke() {
+    ++count_;
+    if ((count_ & (count_ - 1)) == 0) {
+      return true;
+    }
+    return false;
+  }
+
+  static void invokeAction() {
+    if (envoy_bug_failure_record_action_ != nullptr) {
+      envoy_bug_failure_record_action_();
+    }
+  }
+
+private:
+  // This implementation currently only handles one action being set at a time. This is currently
+  // sufficient. If multiple actions are ever needed, the actions should be chained when
+  // additional actions are registered.
+  static std::function<void()> envoy_bug_failure_record_action_;
+  static std::atomic<uint64_t> count_;
```

Comment:
Initialize:

static std::function<void()> envoy_bug_failure_record_action_{};
static std::atomic<uint64_t> count_ = 0;

Reply:
Compiler doesn't like initialized non-const static member variables in line